src/Periphery/ServiceFeeCollector.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.17;
+
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { TransferrableOwnership } from "../Helpers/TransferrableOwnership.sol";
+
+/// @title Service Fee Collector
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for collecting service fees (gas/insurance)
+contract ServiceFeeCollector is TransferrableOwnership {
+    /// Errors ///
+    error TransferFailure();
+    error NotEnoughNativeForFees();
+
+    /// Events ///
+    event GasFeesCollected(
+        address indexed _token,
+        address indexed _receiver,
+        uint256 _feeAmount
+    );
+
+    event InsuranceFeesCollected(
+        address indexed _token,
+        address indexed _receiver,
+        uint256 _feeAmount
+    );
+
+    event FeesWithdrawn(
+        address indexed _token,
+        address indexed _to,
+        uint256 _amount
+    );
+
+    /// Constructor ///
+
+    constructor(address _owner) TransferrableOwnership(_owner) {}
+
+    /// External Methods ///
+
+    /// @notice Collects gas fees
+    /// @param tokenAddress The address of the token to collect
+    /// @param feeAmount The amount of fees to collect
+    /// @param receiver The address to send gas to on the destination chain
+    function collectTokenGasFees(
+        address tokenAddress,
+        uint256 feeAmount,
+        address receiver
+    ) external {
+        LibAsset.depositAsset(tokenAddress, feeAmount);
+        emit GasFeesCollected(tokenAddress, receiver, feeAmount);
```

Comment:
Do we really have to emit an event every time we collect fees? 
Just challenging this because of the latest gas-saving discussions

Reply:
This event is needed in order for our trusted Gas worker to know when to credit gas to someone.